havenapi/actions/keycloak.go

```diff
@@ -0,0 +1,90 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+)
+
+type token struct {
+	Token string `json:"access_token"`
+}
+
+// For example purposes
+type users struct {
+	LastName string `json:"email"`
+	ID       int    `json:"id"`
+}
+
+var adminToken = token{}
+
+// Example env vars (KC_HOST=http://localhost | KC_PORT=2015)
+var adminUser = os.Getenv("KC_ADMIN")
+var adminPw = os.Getenv("KC_PW")
+var keycloakHost = os.Getenv("KC_HOST") + ":" + os.Getenv("KC_PORT")
+var getTokenURL = "/auth/realms/master/protocol/openid-connect/token"
+var getUsersURL = "/auth/admin/realms/havendev/users"
+
+// KeycloakGetToken grabs the token for the admin api.
+func KeycloakGetToken() bool {
```

Comment:
This is great! A question I have is once we get a token, do we need to use it to get refresh tokens or do we just request a new token using the credentials again?

If we just request a new token with the credentials, I wonder if we could easily add a cache with timeout to this function, so that if we called multiple times within say 55 seconds, we would return the existing token, but if the token was about to expire or expired, we would get a fresh token.